featurebyte/service/observation_table.py

```diff
@@ -3,32 +3,136 @@
 """
 from __future__ import annotations
 
-from typing import Any, Dict, Optional, cast
+from typing import Any, Dict, List, Optional, cast
+
+from dataclasses import dataclass
 
 import pandas as pd
 from bson import ObjectId
 
+from featurebyte.api.source_table import SourceTable
+from featurebyte.common.utils import dataframe_from_json
 from featurebyte.enum import DBVarType, MaterializedTableNamePrefix, SpecialColumnName
 from featurebyte.exception import (
     MissingPointInTimeColumnError,
     ObservationTableInvalidContextError,
     UnsupportedPointInTimeColumnTypeError,
 )
 from featurebyte.models.base import FeatureByteBaseDocumentModel
+from featurebyte.models.feature_store import FeatureStoreModel
+from featurebyte.models.materialized_table import ColumnSpecWithEntityId
 from featurebyte.models.observation_table import ObservationTableModel
 from featurebyte.persistent import Persistent
+from featurebyte.query_graph.model.common_table import TabularSource
 from featurebyte.query_graph.node.schema import TableDetails
-from featurebyte.query_graph.sql.materialisation import get_most_recent_point_in_time_sql
+from featurebyte.query_graph.sql.materialisation import (
+    get_earliest_and_most_recent_point_in_time_sql,
+)
+from featurebyte.schema.feature_store import FeatureStoreSample
 from featurebyte.schema.observation_table import ObservationTableCreate, ObservationTableUpdate
 from featurebyte.schema.worker.task.observation_table import ObservationTableTaskPayload
 from featurebyte.service.context import ContextService
 from featurebyte.service.entity import EntityService
 from featurebyte.service.feature_store import FeatureStoreService
 from featurebyte.service.materialized_table import BaseMaterializedTableService
+from featurebyte.service.preview import PreviewService
 from featurebyte.service.session_manager import SessionManagerService
 from featurebyte.session.base import BaseSession
 
 
+@dataclass
+class PointInTimeStats:
+    """
+    Point in time stats
+    """
+
+    earliest: str
+    most_recent: str
+
+
+async def get_point_in_time_stats(
```

Comment:
@YS-L - do you think i could replace this call with the `min TZ offset` and `max TZ offset` stats instead?

Reply:
I don't think so since those are min and max for the TZ offset column of the event timestamp, but here we need the min and max for the point in time column.

Reply:
ah gotcha, what about `top` and `min` stats for the point in time column?

Reply:
`top` is the most frequent value. I think `max` and `min` stats should work.

Reply:
oh indeed, i dunno why i missed the `max` 😓 ok i'll try to swap to those, then all the stats can just be collected in one sql query. (and i won't have to deal with updating the mock too heh)

Reply:
actually i'll fix these in a follow-up so that i can unblock jerome for now!